pkg/dao/repository_configs.go

```diff
@@ -648,6 +649,46 @@ func isTimeout(err error) bool {
 	return false
 }
 
+func (r repositoryConfigDaoImpl) InternalOnly_RefreshRedHatRepo(request api.RepositoryRequest) (*api.RepositoryResponse, error) {
+	newRepoConfig := models.RepositoryConfiguration{}
+	newRepo := models.Repository{}
+
+	request.URL = pointy.Pointer(models.CleanupURL(*request.URL))
+	ApiFieldsToModel(request, &newRepoConfig, &newRepo)
+
+	newRepoConfig.OrgID = config.RedHatOrg
+	newRepo.Origin = config.OriginRedHat
+
+	result := r.db.Clauses(clause.OnConflict{
+		Columns:   []clause.Column{{Name: "url"}},
+		DoUpdates: clause.AssignmentColumns([]string{"origin"})}).Create(&newRepo)
+	if result.Error != nil {
+		return nil, result.Error
+	}
```

Comment:
Does this query mean "if a repository with this URL already exists, rather than error, update its origin column to `red_hat`"?

Reply:
yep!